frontend/src/components/nodes/ValidatorsList.tsx

```diff
@@ -21,11 +21,37 @@ class ValidatorsList extends Component<Props> {
 
     let validatorsList = validators.sort(
       (a: N.ValidationNodeInfo, b: N.ValidationNodeInfo) => {
-        return new BN(b.currentStake).sub(new BN(a.currentStake));
+        // we take "active", "joining", "leaving" validators and sort them firstly
+        // after then we sort the rest
+        const validatingGroup = ["active", "joining", "leaving"];
+
+        const aInValidatingGroup =
+          a.stakingStatus && validatingGroup.indexOf(a.stakingStatus) >= 0;
+        const bInValidatingGroup =
+          b.stakingStatus && validatingGroup.indexOf(b.stakingStatus) >= 0;
+
+        if (aInValidatingGroup && bInValidatingGroup) {
+          return new BN(b.currentStake).cmp(new BN(a.currentStake));
+        } else {
+          const aStake = BN.max(
+            new BN(b.proposedStake || 0),
+            new BN(b.currentStake || 0)
+          );
+          const bStake = BN.max(
+            new BN(a.proposedStake || 0),
+            new BN(a.currentStake || 0)
+          );
+
+          if (aInValidatingGroup || bInValidatingGroup) {
+            return -1;
+          }
+
+          return aStake.cmp(bStake);
+        }
```

Comment:
As per our offline discussion:

* If "a" is a validator, it should be higher on the list
* If "b" is a validator, it should be higher on the list
* otherwise, we compare the stakes (notice that I flipped the cmp order in the last return statement

```suggestion
        if (aInValidatingGroup && bInValidatingGroup) {
          return new BN(b.currentStake).cmp(new BN(a.currentStake));
        } else if (aInValidatingGroup) {
          return -1;
        } else if (bInValidatingGroup) {
          return 1;
        } else {
          const aStake = BN.max(
            new BN(b.proposedStake || 0),
            new BN(b.currentStake || 0)
          );
          const bStake = BN.max(
            new BN(a.proposedStake || 0),
            new BN(a.currentStake || 0)
          );
          return bStake.cmp(aStake);
        }
```